spyderlib/workers/updates.py

```diff
@@ -0,0 +1,148 @@
+    """
+    A QMessageBox derived widget that includes a QCheckBox under the message
+    and on top of the buttons.
+    """
+    def __init__(self, *args, **kwargs):
+        super(MessageCheckBox, self).__init__(*args, **kwargs)
+
+        self._checkbox = QCheckBox()
+
+        # Access the Layout of the MessageBox to add the Checkbox
+        layout = self.layout()
+        layout.addWidget(self._checkbox, 1, 1, Qt.AlignRight)
+
+    # Methods to access the checkbox
+    def is_checked(self):
+        return self._checkbox.isChecked()
+
+    def set_checked(self, value):
+        return self._checkbox.setChecked(value)
+
+    def set_check_visible(self, value):
+        self._checkbox.setVisible(value)
+
+    def is_check_visible(self):
+        self._checkbox.isVisible()
+
+    def checkbox_text(self):
+        self._checkbox.text()
+
+    def set_checkbox_text(self, text):
+        self._checkbox.setText(text)
+
+
+class WorkerUpdates(QObject):
+    """
+    Worker that checks for releases using the Github API without blocking the
+    Spyder user interface, in case of connections issues.
+    """
+    sig_ready = Signal()
+
+    def __init__(self, parent, feedback):
+        QObject.__init__(self)
+        self._parent = parent
+        self.feedback = feedback
+        self.error = feedback
+        self.latest_release = None
+
+    def is_stable_version(self, version):
+        """
+        A stable version has no letters in the final part, it has only numbers.
+
+        Stable version example: 1.2, 1.3.4, 1.0.5
+        Not stable version: 1.2alpha, 1.3.4beta, 0.1.0rc1, 3.0.0dev
+        """
+        if not isinstance(version, tuple):
+            version = version.split('.')
+        last_part = version[-1]
+
+        try:
+            int(last_part)
+            return True
+        except ValueError:
+            return False
+
+    def check_update_available(self, version, releases):
+        """ """
+        from spyderlib.utils.programs import check_version
+
+        if self.is_stable_version(version):
+            # Remove non stable versions from the list
+            releases = [r for r in releases if self.is_stable_version(r)]
+
+        latest = releases[0]
+        latest_release = releases[0]
+
+        if version.endswith('dev'):
+            return (False, latest_release)
+
+        if self.is_stable_version(latest) and \
+                version.startswith(latest) and latest != version:
+            # Modify the last part so that i.e. 3.0.0 is bigger than 3.0.0rc2
+            parts = latest.split('.')
+            parts = parts[:-1] + [parts[-1] + 'z']
+            latest = '.'.join(parts)
+
+        return (check_version(version, latest, '<'), latest_release)
+
+    def start(self):
+        """ """
+        if PY3:
+            from urllib.request import urlopen
+            from urllib.error import URLError, HTTPError
+        else:
+            from urllib2 import urlopen, URLError, HTTPError
```

Comment:
Actually LooseVersion works pretty well, except in this cases... where it does not understand that 

3.0.0 is bigger than 3.0.0rc1 etc... so the "hack" works pretty well within the context of loose version.

http://epydoc.sourceforge.net/stdlib/distutils.version.LooseVersion-class.html